services/sts/src/main/java/software/amazon/awssdk/services/sts/internal/StsWebIdentityTokenFileCredentialsProvider.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.services.sts.internal;
+
+import static software.amazon.awssdk.utils.StringUtils.trim;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Optional;
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.auth.credentials.AwsCredentials;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.internal.WebIdentityTokenCredentialProperties;
+import software.amazon.awssdk.core.SdkSystemSetting;
+import software.amazon.awssdk.services.sts.StsClient;
+import software.amazon.awssdk.utils.ToString;
+
+
+/**
+ * A credential provider that will read web identity token file path, aws role arn, aws session name
+ * from system properties or environment variables  and StsClient for using
+ * web identity token credentials with STS. If StsClient is not passed or null then it will use Default Sts client.
+ */
+@SdkPublicApi
+public class StsWebIdentityTokenFileCredentialsProvider implements AwsCredentialsProvider {
```

Comment:
Should this extend StsCredentialsProvider like StsAssumeRoleWithWebIdentityCredentialsProvider does?

Reply:
doen , new changes in StsWebIdentityTokenFileCredentialsProvider